packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/IntentSender.java

```diff
@@ -158,6 +169,45 @@ Intent buildIntent(
         intent.setPackage(null);
       }
     }
+    Boolean needIgnore = ignoredPackages != null && ignoredPackages.size() > 0;
+    if (needIgnore || showChooser) {
+      PackageManager packageManager;
+      if (activity != null) {
+        packageManager = activity.getPackageManager();
+      } else {
+        packageManager = applicationContext.getPackageManager();
+      }
+      List<ResolveInfo> activities = packageManager.queryIntentActivities(intent, 0);
+      ArrayList<Intent> targetIntents = new ArrayList<Intent>();
+
+      if (needIgnore) {
+        for (ResolveInfo currentInfo : activities) {
+          if (ignoredPackages.indexOf(currentInfo.activityInfo.packageName) == -1) {
+            Intent targetIntent = new Intent(intent);
+            targetIntent.setPackage(currentInfo.activityInfo.packageName);
+            targetIntents.add(targetIntent);
+          }
+        }
+      }
+
+      if (!showChooser) {
+        String packageToLaunch;
+        try {
+          if (ignoredPackages != null) {
+            packageToLaunch = targetIntents.get(0).getPackage();
+          } else {
+            packageToLaunch = activities.get(0).activityInfo.packageName;
+          }
+          intent.setPackage(packageToLaunch);
+          Log.v(TAG, "Resolve this intent with package: " + packageToLaunch);
+        } catch (IndexOutOfBoundsException exception) {
+          Log.v(TAG, "Not found packages to resolve this intent: " + exception);
+        }
+      } else {
+        intent = Intent.createChooser(targetIntents.remove(0), "");
```

Comment:
I have 2 concerns on this line. First, I don't thin we should be replacing the `intent` value. It is very easy for someone to miss this line and believe that `intent` always refers to the same instance throughout this method.

Second, is there a reason that `remove(0)` is being called? If not, please consider changing it to `get(0)`. If there is a reason, then please consider moving that call to its own line and commenting as to why we need `remove()` instead of `get()`.

Reply:
I call `remove(0)` because I then call the `putExtra` method with the entire array of packages. If I do not remove the first intent, then the corresponding package will be in the chooser twice. I could call `createChooser` with `new Intent()` as the first argument, but I'm not sure that is good.

Reply:
It also took me a bit of reading to figure out why we were popping the head from the list here and if it was a mistake. I think the decision makes sense, but an implementation comment would be helpful.